sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/DefaultIndexedFile.java

```diff
@@ -38,37 +37,45 @@
  */
 @Immutable
 public class DefaultIndexedFile extends DefaultInputComponent implements IndexedFile {
-  private final String relativePath;
+  private final String projectRelativePath;
+  private final String moduleRelativePath;
   private final String moduleKey;
-  private final Path moduleBaseDir;
   private final String language;
   private final Type type;
-  private final Path path;
+  private final Path absolutePath;
+  private final SensorStrategy sensorStrategy;
 
   /**
    * Testing purposes only!
    */
-  public DefaultIndexedFile(String moduleKey, Path moduleBaseDir, String relativePath, @Nullable String language) {
-    this(moduleKey, moduleBaseDir, relativePath, language, TestInputFileBuilder.nextBatchId());
-  }
-
-  public DefaultIndexedFile(String moduleKey, Path moduleBaseDir, String relativePath, @Nullable String language, int batchId) {
-    this(moduleKey, moduleBaseDir, relativePath, Type.MAIN, language, batchId);
+  public DefaultIndexedFile(String moduleKey, Path baseDir, String relativePath, @Nullable String language) {
+    this(baseDir.resolve(relativePath), moduleKey, PathUtils.sanitize(relativePath), PathUtils.sanitize(relativePath), Type.MAIN, language, TestInputFileBuilder.nextBatchId(),
+      new SensorStrategy());
   }
 
-  public DefaultIndexedFile(String moduleKey, Path moduleBaseDir, String relativePath, Type type, @Nullable String language, int batchId) {
+  public DefaultIndexedFile(Path absolutePath, String moduleKey, String projectRelativePath, String moduleRelativePath, Type type, @Nullable String language, int batchId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)